src/cmd/services/m3coordinator/ingest/carbon/ingest.go

```diff
@@ -115,17 +137,22 @@ func NewIngester(
 			opts.InstrumentOptions.MetricsScope()),
 
 		rules: compiledRules,
+
+		lineResourcesPool: resourcePool,
 	}, nil
 }
 
 type ingester struct {
+	ctx                  context.Context
```

Comment:
Can we make this per connection, e.g. at top of `Handle(...)` we have a `ctx := context.Background()`?  That way at least each connection gets its own unique context.  (And there's not likely to be more than hundreds of connections, so that should avoid it being alloc heavy).